torch/_inductor/codegen/triton.py

```diff
@@ -2827,6 +3083,9 @@ def codegen_kernel(self, name=None):
                 continue
             argdefs.append(f"{tree.prefix.upper()}BLOCK : tl.constexpr")
 
+        if self.cooperative_reduction:
+            argdefs.append("RSPLIT : tl.constexpr")
```

Comment:
How will this work with dynamic shapes ?

Reply:
This should work with dynamic shapes already.  Each SM processes `rnumel // RSPLIT` elements.  RSPLIT is a constant, but rnumel is not.